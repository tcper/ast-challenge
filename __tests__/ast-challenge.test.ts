import * as ts from "typescript";
import generateFunction from '../src/index'


it('works', () => {
    const input = {
        "Pools": {
            "requestType": "QueryPoolsRequest",
            "responseType": "QueryPoolsResponse",
            "interfaceName": "UsePoolsQuery",
            "queryServiceName": "queryService",
            "queryServiceFunction": "pools",
            "keyName": "poolsQuery"
        }
    }
    let ast = generateFunction(input["Pools"]);
    const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed });
    let code = ast.map((line) => printer.printNode(ts.EmitHint.Unspecified, line, ts.createSourceFile("source.ts", "", ts.ScriptTarget.ESNext, false, ts.ScriptKind.TS))).join("\n")
    expect(code).toMatchSnapshot()
});