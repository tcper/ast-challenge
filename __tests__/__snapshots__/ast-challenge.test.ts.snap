// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
    request?: QueryPoolsRequest;
}
const usePools = <TData = QueryPoolsResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;
